assets/js/base/hooks/shipping/use-shipping-rates.js

```diff
@@ -1,64 +1,33 @@
 /**
  * External dependencies
  */
-import { useSelect, useDispatch } from '@wordpress/data';
-import { useReducer, useEffect } from '@wordpress/element';
-import isShallowEqual from '@wordpress/is-shallow-equal';
-import { useDebounce } from 'use-debounce';
+import { useSelect } from '@wordpress/data';
 import { CART_STORE_KEY as storeKey } from '@woocommerce/block-data';
 
 /**
  * Internal dependencies
  */
 import { useStoreCart } from '../cart/use-store-cart';
-import { pluckAddress } from '../../utils';
+import { useShippingAddress } from '../shipping/use-shipping-address';
 /**
  * This is a custom hook that is wired up to the `wc/store/cart/shipping-rates` route.
  * Given a a set of default fields keys, this will handle shipping form state and load
  * new rates when certain fields change.
  *
- * @param {Array} addressFieldsKeys an array containing default fields keys.
- *
  * @return {Object} This hook will return an object with three properties:
  *                 - {Boolean} shippingRatesLoading A boolean indicating whether the shipping
  *                   rates are still loading or not.
  *                 - {Function} setShippingAddress  An function that optimistically
  *                   update shipping address and dispatches async rate fetching.
  *                 - {Object} shippingAddress       An object containing shipping address.
  */
-export const useShippingRates = ( addressFieldsKeys ) => {
+export const useShippingRates = () => {
 	const { cartErrors, shippingRates } = useStoreCart();
-	const addressFields = Object.fromEntries(
-		addressFieldsKeys.map( ( key ) => [ key, '' ] )
-	);
-	const derivedAddress = shippingRates[ 0 ]?.destination;
```

Comment:
I do agree that this is fragile, when I first implemented it, I didn’t find any other solutions, exposing yet other values from the cart endpoint didn’t feel right to me, I feel we're shipping too much data from that endpoint.